### PR TITLE
main/libssh: upgrade to 0.8.7

### PR DESCRIPTION
**Pull request by TBK**

### CODE REVIEW

File: main/libssh/APKBUILD [hunk 1]
```diff
@@ -1,32 +1,40 @@
...
-license="LGPL"
-makedepends="zlib-dev openssl-dev cmake doxygen"
+license="LGPL-2.1-or-later BSD-2-Clause"
+makedepends="cmake doxygen openssl-dev zlib-dev"
+checkdepends="cmocka-dev"
 subpackages="$pkgname-dev"
-options="!check"
-source="https://www.libssh.org/files/0.7/libssh-$pkgver.tar.xz
-	fix-includes.patch"
+source="https://www.libssh.org/files/${pkgver:0:-2}/libssh-$pkgver.tar.xz"
```
Comment (tcely) [inline main/libssh/APKBUILD]: This is a bashism we don't want.

File: main/libssh/APKBUILD [hunk 2]
```diff
@@ -1,32 +1,40 @@
...
 builddir="$srcdir"/$pkgname-$pkgver
```
Comment (tcely) [inline main/libssh/APKBUILD]:
```suggestion
```

File: main/libssh/APKBUILD (hunk 1 above)
Comment (TBK) [reply main/libssh/APKBUILD]: Thanks for the suggestion. I had already made the change locally but I was waiting for upstream to provide a fix for the failing test. They have now fix the issue and I have update the PR so it should be ready to go.

File: main/libssh/APKBUILD [hunk 3]
```diff
@@ -1,32 +1,43 @@
...
-pkgrel=1
+pkgver=0.8.7
+pkgrel=0
 pkgdesc="Library for accessing ssh client services through C libraries"
-url="http://www.libssh.org/"
+url="https://www.libssh.org"
 arch="all"
-license="LGPL"
-makedepends="zlib-dev openssl-dev cmake doxygen"
+license="LGPL-2.1-or-later BSD-2-Clause"
```
Comment (tcely) [inline main/libssh/APKBUILD]: A boolean like AND / OR would clarify the license situation.

File: main/libssh/APKBUILD [hunk 4]
```diff
@@ -1,32 +1,43 @@
...
+makedepends="cmake doxygen openssl-dev zlib-dev"
```
Comment (tcely) [inline main/libssh/APKBUILD]: I wonder about moving `openssl-dev` and `zlib-dev` to `depends_dev`.

**Comment by TBK** [comment]
I have bumped the dependent aports since 0.7.8 is 99.23% BC. A so file and 2 symbols were removed - https://abi-laboratory.pro/index.php?view=objects_report&l=libssh&v1=0.7.7&v2=0.8.0

**Comment by andypost** [comment]
https://www.libssh.org/2019/06/28/libssh-0-9-0/

**Comment by maxice8** [comment]
libssh is already on 0.8.7

**Pull request closed without merging**